MudAutocomplete: Remove SearchFuncWithCancel, Add CancellationToken to SearchFunc

jperson2000 opened the pull request:
This breaking change removes the `SearchFuncWithCancel` method and moves the `CancellationToken` functionality to the `SearchFunc` method.  

## Description

Recent changes to the MudTable `ServerData` method ([PR 8407](https://github.com/MudBlazor/MudBlazor/pull/8407)) led to a desire to use the same single-method pattern for other cancelable methods.  This update:

* Removes the `SearchFuncWithCancel` method
* Moves the `CancellationToken` parameter to the `SearchFunc` method
* Updates all tests and examples to include the new `CancellationToken` parameter.

## How Has This Been Tested?

Existing tests for `MudAutocomplete` were modified to include the `CancellationToken` parameter.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] The PR is submitted to the correct branch (`dev`).
- [X] My code follows the code style of this project.
- [X] I've added relevant tests.

## Notes for Reviewers

This should probably get a `v7` tag and be released with similar breaking changes to `MudTable` in the works for #8447

Commit:
MudAutocomplete: Removed SearchFuncWithCancel and added CancellationToken to SearchFunc

Commit:
MudAutocomplete: Removed SearchFuncWithCancel and added CancellationToken to SearchFunc

Commit:
MudAutocomplete: Removed SearchFuncWithCancel and added CancellationToken to SearchFunc

Commit:
MudAutocomplete: Removed SearchFuncWithCancel and added CancellationToken to SearchFunc

Commit:
MudAutocomplete: Removed SearchFuncWithCancel and added CancellationToken to SearchFunc

Commit:
MudAutocomplete: Removed SearchFuncWithCancel and added CancellationToken to SearchFunc

Commit:
MudAutocomplete: Removed SearchFuncWithCancel and added CancellationToken to SearchFunc

Commit:
Merge branch 'feature/mudautocomplete-searchfunc-cancellationtoken' of https://github.com/jperson2000/MudBlazor into feature/mudautocomplete-searchfunc-cancellationtoken

Commit:
Merge branch 'feature/mudautocomplete-searchfunc-cancellationtoken' of https://github.com/jperson2000/MudBlazor into feature/mudautocomplete-searchfunc-cancellationtoken

henon commented:
There are conflicts. Please merge dev to resolve them.

Added to v7.0.0 Migration Guide #8447

danielchalmers committed:
Docs: Fix binding regressions in examples (#8494)

ArieGato committed:
Dialog: Rename ClassContent and ClassActions parameters (#8481)

Co-authored-by: Artyom M <artem.melnikov@live.com>

ScarletKuro commented:
LGTM, just need to rebase and fix conflicts like henon said.

Commit:
Merge branch 'feature/mudautocomplete-searchfunc-cancellationtoken' of https://github.com/jperson2000/MudBlazor into feature/mudautocomplete-searchfunc-cancellationtoken

Commit:
Merge branch 'feature/mudautocomplete-searchfunc-cancellationtoken' of https://github.com/jperson2000/MudBlazor into feature/mudautocomplete-searchfunc-cancellationtoken

Commit:
Merge branch 'feature/mudautocomplete-searchfunc-cancellationtoken' of https://github.com/jperson2000/MudBlazor into feature/mudautocomplete-searchfunc-cancellationtoken

Commit:
MudAutocomplete: Rebased to dev

Commit:
MudAutocomplete: Re-added CancellationToken to search methods after rebasing

jperson2000 commented:
Alright, thanks again for all the reviews and feedback!  I've rebased onto `dev` and added 1 commit to re-implement the new `token` parameter in `MudBlazor.Docs\Shared\AppBar.razor`.

henon commented:
Next time please just merge `dev`, somehow rebasing with Visual Studio messes up the PR. You are done, however, so it doesn't really matter. Just so you know.

henon commented:
And thanks for the contribution @jperson2000 !
